documentation/general/program-scale-testing-requirements/hero-implementation-details.md

```diff
@@ -0,0 +1,247 @@
+- IEEE defaults autoneg is preferable but at a minimum if AN is disabled please ensure FEC is enabled. With FEC disabled we observed few packet drops in the DACs and that can create a lot of hasle hunting down a lost packet that has nothing to do with DASH performance.  
+
+#### Testbed examples
+
+Few examples bellow with 100G cables, with 400G cables with fanout cables, single DPU or applaince or smartswitch.
+
+testbed1 
+
+![testbed1](./testbed1.svg)
+
+testbed2 
+
+![testbed2](./testbed2.svg)
+
+testbed3 
+
+![testbed3](./testbed3.svg)
+
+
+>**NOTE**: At this point all hardware should be in the lab powered on, accesible via IP and have link up on all the interfaces.
+
+
+### "1 IP" test
+
+Minimum config posible to run traffic through. 
+
+#### Objectives
+
+##### validate the hardware and software. 
+
+It ensures we can program the DPU via private API, SAI or DASH and that we can pass 1 packets end to end from traffic generator through the device under test and back.
+
+##### can also provide best case scenario performance numbers
+
+its a maybe because 1 packets replicated milions of times may not necesarly work best for all hardware implementations.
+
+
+### Baby Hero test
+
+It is ment to be an intermediary step between 1 IP test and Hero test. 
+
+We try to keep the scale at ~1% of hero test by using only one prefix per ACL instead of 100
+
+| Metric                       | Baby Hero | % of HERO  |
+|------------------------------|-----------|------------|
+| VNIs                         | 32        | 3%         |
+| ENIs                         | 32        | 100%       |
+| NSGs per ENI                 | 10        | 100%       |
+| NSGs                         | 320       | 100%       |
+| ACL rules per NSG            |  1,000    | 100%       |
+| ACL rules                    |  320,000  | 100%       |
+| ACL prefixes per ACL rule    | 1         | 1%         |
+| ACL prefixes                 |  320,000  | 1%         |
+| ACL ports                    | NA        | 0%         |
+| VNETs (Outbound)             | 32        | 100%       |
+| Outbound routes per ENI      | 5K        |            |
+| Outbound routes              | 160K      | 5%         |
+| CA-to-PA mapping (Outbound)  | 80320     | 1%         |
+| Inbound Routes per ENI       | 1         | 0%         |
+| Inbound Routes               | 32        | 100%       |
+| PAs (Inbound PA Validation)  | 32        | 100%       |
+
+
+### In between scale
+
+If Hero test scale numbers cannot be met we can add another checkpoint to gather aditional data before final implementation is ready
+
+### Hero test
+
+The test exactly as specified in the DASH requirements.
+
+### Best case scenario
+
+If any of the previous tests have not shown the best case scenario we can run a test with the best case scenario in mind.
+
+### Worst case scenario
+
+If any of the previous tests have not shown the worst case scenario we add this test as well (without exceeding hero test scale).
```

Comment:
I don't know what this means.

Reply:
rephrased "If we can find a scenario where we obtain lower performance numbers then the numbers previously obtained during (1ip, baby hero, hero test ...) this will be added as a new data point to the results."